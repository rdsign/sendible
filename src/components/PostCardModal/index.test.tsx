import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { PostCardModal } from './index';
import { Template } from '../../../spec/template';

describe('PostCardModal', () => {
    const onClose = jest.fn();
    const onError = jest.fn();
    const onSave = jest.fn();

    afterEach(() => {
        jest.clearAllMocks();
    });

    const component = (
        <Template>
            <PostCardModal
                date="01/02/202"
                id="c020fcff-11e3-480d-9a6d-6ebb5772dd0d"
                imageUrl="https://images.unsplash.com/photo-1499781350541-7783f6c6a0c8?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1845&q=80"
                message="Lorem ipsum dolor sit amet, consectetur adipiscing elit."
                time="16:3"
                isVisible
                close={onClose}
                error={onError}
                save={onSave}
            />
        </Template>
    );

    test('should match snapshot', () => {
        const snapshot = create(component);
        expect(snapshot).toMatchSnapshot();
    });

    test('message input field should match the passed prop and changes', () => {
        render(component);
        const [textarea] = screen.getAllByRole('textbox');

        userEvent.type(textarea, ' Nice!');

        expect(textarea).toHaveValue(
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nice!',
        );
    });

    test('should call onError upon click on save button, because date and time are incorrect', () => {
        render(component);
        const [button] = screen.getAllByRole('button');

        userEvent.click(button);

        expect(onError).toHaveBeenCalledTimes(1);
    });

    test('should call onClose upon click on close button', () => {
        render(component);
        const [, button] = screen.getAllByRole('button');

        userEvent.click(button);

        expect(onClose).toHaveBeenCalledTimes(1);
    });

    test('should call onClose upon click on overlay', () => {
        render(component);
        const overlay = screen.getByTestId('overlay');

        userEvent.click(overlay);

        expect(onClose).toHaveBeenCalledTimes(1);
    });

    test('date and time inputs should format the value based on date/time masks', () => {
        render(component);
        const [, date, time] = screen.getAllByRole('textbox');

        userEvent.type(date, '9000');
        userEvent.type(time, '0000');
        expect(date).toHaveValue('01/02/2029');
        expect(time).toHaveValue('16:30');
    });
});
