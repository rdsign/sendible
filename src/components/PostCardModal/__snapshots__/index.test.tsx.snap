// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCardModal should match snapshot 1`] = `
.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #4d36ff;
  border-radius: 5rem;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.6rem;
  padding: 1.4rem 2.6rem;
  -webkit-transition: all 1s cubic-bezier(0.23,1,0.32,1);
  transition: all 1s cubic-bezier(0.23,1,0.32,1);
}

.c9 svg {
  fill: #ffffff;
  height: auto;
  margin-right: 1rem;
  opacity: 0.5;
  width: 1.8rem;
}

.c9:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border-radius: 5rem;
  color: #201e24;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.6rem;
  padding: 1.4rem 2.6rem;
  -webkit-transition: all 1s cubic-bezier(0.23,1,0.32,1);
  transition: all 1s cubic-bezier(0.23,1,0.32,1);
}

.c10 svg {
  fill: #201e24;
  height: auto;
  opacity: 1;
  width: 1.8rem;
}

.c10:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-backdrop-filter: blur(3rem);
  backdrop-filter: blur(3rem);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  position: fixed;
  top: -101%;
  -webkit-transition: all 1s cubic-bezier(0.23,1,0.32,1);
  transition: all 1s cubic-bezier(0.23,1,0.32,1);
  visibility: hidden;
  width: 100%;
  z-index: 100;
  top: 0;
  visibility: visible;
}

.c1 {
  background: #ffffff;
  border-radius: 0.8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6rem 6rem 4rem;
  position: relative;
  width: 100%;
}

.c1 .close-button {
  padding: 3rem;
  position: absolute;
  right: 3rem;
  top: 3rem;
}

.c2 {
  width: 94%;
}

.c2 textarea {
  border: 0;
  color: #201e24;
  font-size: 2rem;
  height: 11rem;
  line-height: 1.66em;
  margin-bottom: 4rem;
  resize: none;
  width: 100%;
}

.c2 textarea::-webkit-input-placeholder {
  color: #a09fa0;
}

.c2 textarea::-moz-placeholder {
  color: #a09fa0;
}

.c2 textarea:-ms-input-placeholder {
  color: #a09fa0;
}

.c2 textarea::placeholder {
  color: #a09fa0;
}

.c3 {
  border-radius: 0.8rem;
  height: 15rem;
  overflow: hidden;
  width: 40rem;
}

.c3 img {
  height: 100%;
  object-fit: cover;
  width: 100%;
}

.c4 {
  background: #f4f4f4;
  height: 1px;
  margin: 6rem 0 4rem;
  margin-left: -6rem;
  width: calc(100% + 12rem);
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: 6rem;
}

.c7 svg {
  fill: #4d36ff;
  width: 3.5rem;
}

.c7 input {
  color: #4d36ff;
  font-size: 1.6rem;
  font-weight: 600;
  margin-left: 1rem;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  width: 10rem;
}

.c8 {
  cursor: pointer;
  overflow: hidden;
  position: relative;
}

.c8 input[type='file'] {
  cursor: pointer;
  height: 100%;
  left: 0;
  margin: 0;
  opacity: 0;
  position: absolute;
  top: 0;
  width: 100%;
}

.c11 {
  background: #201e24;
  cursor: pointer;
  height: 100%;
  left: 0;
  opacity: 0.9;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: -1;
}

@media all and (min-width:260px) and (max-width:767px) {
  .c1 {
    margin: 0;
    min-width: 100vw;
    padding: 3rem 0 0;
  }

  .c1 .close-button {
    right: 0;
    top: -5.5rem;
  }

  .c1 .close-button svg {
    fill: white;
  }
}

@media all and (min-width:260px) and (max-width:767px) {
  .c2 {
    margin: 0 auto;
    width: 85%;
  }

  .c2 textarea {
    font-size: 1.8rem;
    height: 25rem;
  }
}

@media all and (min-width:768px) and (max-width:1024px) {
  .c2 {
    width: 90%;
  }

  .c2 textarea {
    height: 21rem;
  }
}

@media all and (min-width:260px) and (max-width:767px) {
  .c3 {
    height: 16vh;
    width: 100%;
  }
}

@media all and (min-width:260px) and (max-width:767px) {
  .c4 {
    margin-bottom: 3rem;
    margin-top: 3rem;
  }
}

@media all and (min-width:260px) and (max-width:767px) {
  .c5 {
    bottom: 0;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    left: 0;
    width: 100%;
  }

  .c5 button {
    border-radius: 0;
    font-size: 2rem;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    margin-top: 2rem;
    padding: 3rem 0;
    width: 100%;
  }
}

@media all and (min-width:260px) and (max-width:767px) {
  .c6 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    margin: 0 auto;
    width: 85%;
  }
}

@media all and (min-width:260px) and (max-width:767px) {
  .c7 {
    margin-right: 0;
  }

  .c7 input {
    font-size: 1.8rem;
  }

  .c7 #date {
    width: 11.2rem;
  }

  .c7 #time {
    width: 5.2rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1 wrapper"
  >
    <div
      className="c2"
    >
      <textarea
        maxLength={280}
        onChange={[Function]}
        placeholder="Write your message here..."
        value="Lorem ipsum dolor sit amet, consectetur adipiscing elit."
      />
      <div
        className="c3"
      >
        <img
          alt="post"
          src="https://images.unsplash.com/photo-1499781350541-7783f6c6a0c8?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1845&q=80"
        />
      </div>
    </div>
    <div
      className="c4"
    />
    <div
      className="c5 actions"
    >
      <div
        className="c6"
      >
        <div
          className="c7"
        >
          <div
            className="c8"
          >
            <svg
              data-testid="svg"
              fill="inherit"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M3.26531 20C3.26531 10.7577 10.7577 3.26531 20 3.26531C29.2423 3.26531 36.7347 10.7577 36.7347 20C36.7347 29.2423 29.2423 36.7347 20 36.7347C10.7577 36.7347 3.26531 29.2423 3.26531 20ZM20 0C8.95431 0 0 8.95431 0 20C0 31.0457 8.95431 40 20 40C31.0457 40 40 31.0457 40 20C40 8.95431 31.0457 0 20 0ZM21.6327 9.79592C21.6327 8.89423 20.9017 8.16327 20 8.16327C19.0983 8.16327 18.3673 8.89423 18.3673 9.79592V20C18.3673 20.433 18.5394 20.8483 18.8455 21.1545L24.968 27.2769C25.6056 27.9145 26.6393 27.9145 27.2769 27.2769C27.9145 26.6393 27.9145 25.6056 27.2769 24.968L21.6327 19.3237V9.79592Z"
                fillRule="evenodd"
              />
            </svg>
            <input
              accept="image/png, image/jpeg"
              id="avatar"
              name="avatar"
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
        <div
          className="c7"
        >
          <svg
            data-testid="svg"
            fill="inherit"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M3.26531 20C3.26531 10.7577 10.7577 3.26531 20 3.26531C29.2423 3.26531 36.7347 10.7577 36.7347 20C36.7347 29.2423 29.2423 36.7347 20 36.7347C10.7577 36.7347 3.26531 29.2423 3.26531 20ZM20 0C8.95431 0 0 8.95431 0 20C0 31.0457 8.95431 40 20 40C31.0457 40 40 31.0457 40 20C40 8.95431 31.0457 0 20 0ZM21.6327 9.79592C21.6327 8.89423 20.9017 8.16327 20 8.16327C19.0983 8.16327 18.3673 8.89423 18.3673 9.79592V20C18.3673 20.433 18.5394 20.8483 18.8455 21.1545L24.968 27.2769C25.6056 27.9145 26.6393 27.9145 27.2769 27.2769C27.9145 26.6393 27.9145 25.6056 27.2769 24.968L21.6327 19.3237V9.79592Z"
              fillRule="evenodd"
            />
          </svg>
          <input
            id="date"
            maxLength={10}
            onChange={[Function]}
            placeholder="Date"
            type="input"
          />
        </div>
        <div
          className="c7"
        >
          <svg
            data-testid="svg"
            fill="inherit"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M3.26531 20C3.26531 10.7577 10.7577 3.26531 20 3.26531C29.2423 3.26531 36.7347 10.7577 36.7347 20C36.7347 29.2423 29.2423 36.7347 20 36.7347C10.7577 36.7347 3.26531 29.2423 3.26531 20ZM20 0C8.95431 0 0 8.95431 0 20C0 31.0457 8.95431 40 20 40C31.0457 40 40 31.0457 40 20C40 8.95431 31.0457 0 20 0ZM21.6327 9.79592C21.6327 8.89423 20.9017 8.16327 20 8.16327C19.0983 8.16327 18.3673 8.89423 18.3673 9.79592V20C18.3673 20.433 18.5394 20.8483 18.8455 21.1545L24.968 27.2769C25.6056 27.9145 26.6393 27.9145 27.2769 27.2769C27.9145 26.6393 27.9145 25.6056 27.2769 24.968L21.6327 19.3237V9.79592Z"
              fillRule="evenodd"
            />
          </svg>
          <input
            id="time"
            maxLength={5}
            onChange={[Function]}
            placeholder="Time"
            type="input"
          />
        </div>
      </div>
      <button
        className="c9"
        color="white"
        label="Save"
        onClick={[Function]}
        type="button"
      >
        <svg
          data-testid="svg"
          fill="inherit"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M3.26531 20C3.26531 10.7577 10.7577 3.26531 20 3.26531C29.2423 3.26531 36.7347 10.7577 36.7347 20C36.7347 29.2423 29.2423 36.7347 20 36.7347C10.7577 36.7347 3.26531 29.2423 3.26531 20ZM20 0C8.95431 0 0 8.95431 0 20C0 31.0457 8.95431 40 20 40C31.0457 40 40 31.0457 40 20C40 8.95431 31.0457 0 20 0ZM21.6327 9.79592C21.6327 8.89423 20.9017 8.16327 20 8.16327C19.0983 8.16327 18.3673 8.89423 18.3673 9.79592V20C18.3673 20.433 18.5394 20.8483 18.8455 21.1545L24.968 27.2769C25.6056 27.9145 26.6393 27.9145 27.2769 27.2769C27.9145 26.6393 27.9145 25.6056 27.2769 24.968L21.6327 19.3237V9.79592Z"
            fillRule="evenodd"
          />
        </svg>
        Save
      </button>
    </div>
    <button
      className="c10 close-button"
      color="gray800"
      onClick={[Function]}
      title="close"
      type="button"
    >
      <svg
        data-testid="svg"
        fill="inherit"
        height="40"
        viewBox="0 0 40 40"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M3.26531 20C3.26531 10.7577 10.7577 3.26531 20 3.26531C29.2423 3.26531 36.7347 10.7577 36.7347 20C36.7347 29.2423 29.2423 36.7347 20 36.7347C10.7577 36.7347 3.26531 29.2423 3.26531 20ZM20 0C8.95431 0 0 8.95431 0 20C0 31.0457 8.95431 40 20 40C31.0457 40 40 31.0457 40 20C40 8.95431 31.0457 0 20 0ZM21.6327 9.79592C21.6327 8.89423 20.9017 8.16327 20 8.16327C19.0983 8.16327 18.3673 8.89423 18.3673 9.79592V20C18.3673 20.433 18.5394 20.8483 18.8455 21.1545L24.968 27.2769C25.6056 27.9145 26.6393 27.9145 27.2769 27.2769C27.9145 26.6393 27.9145 25.6056 27.2769 24.968L21.6327 19.3237V9.79592Z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </div>
  <div
    className="c11"
    data-testid="overlay"
    onClick={[Function]}
  />
</div>
`;
